app/src/main/java/org/asteroidos/sync/ble/TimeService.java

```diff
@@ -34,26 +39,43 @@
     public static final String PREFS_NAME = "TimePreference";
     public static final String PREFS_SYNC_TIME = "syncTime";
     public static final boolean PREFS_SYNC_TIME_DEFAULT = true;
+    public static final String TIME_SYNC_INTENT = "org.asteroidos.sync.TIME_SYNC_REQUEST_LISTENER";
 
     private BleDevice mDevice;
+    private Context mCtx;
 
     private SharedPreferences mTimeSyncSettings;
 
+    private TimeSyncReqReceiver mSReceiver;
+    private PendingIntent alarmPendingIntent;
+    private AlarmManager alarmMgr;
+
     public TimeService(Context ctx, BleDevice device) {
         mDevice = device;
+        mCtx = ctx;
         mTimeSyncSettings = ctx.getSharedPreferences(PREFS_NAME, 0);
         mTimeSyncSettings.registerOnSharedPreferenceChangeListener(this);
     }
 
     public void sync() {
+        updateTime();
```

Comment:
I totally understand your point with removing the delayed Runnable but unfortunately this is one of those nasty hack I had to put in the codebase to workaround BLE issues
https://github.com/AsteroidOS/AsteroidOSSync/commit/e01935731d13ce9694cc5c4aff1e8107cc4565c2
I'm afraid this specific change could lead to regressions.